pkg/provisioner/ironic/ironic.go

```diff
@@ -1743,42 +1743,67 @@ func (p *ironicProvisioner) PowerOn() (result provisioner.Result, err error) {
 
 	if ironicNode.PowerState != powerOn {
 		if ironicNode.TargetPowerState == powerOn {
```

Comment:
Target states means something is in progress, you won't get a target power state set with last error.

Reply:
Thanks for that. Fixed it.